core/src/library.rs

```diff
@@ -90,6 +90,25 @@ impl<'gc> Avm2ClassRegistry<'gc> {
         movie: Arc<SwfMovie>,
         symbol: CharacterId,
     ) {
+        if let Some(old) = self.class_map.get(&class_object) {
+            // We should never have multiple distinct symbols in the same movie linked
+            // with the same class.
+            // However, flash player *does* allow duplicate SymbolClass entires in the same
+            // movie - the same (symbol, class) pair can show up across multiple SymbolClass
+            // tags.
+            if Arc::ptr_eq(&movie, &old.0) && symbol != old.1 {
+                panic!(
+                    "Tried to overwrite class {:?} id={:?} with symbol id={:?} from same movie",
+                    class_object, old.1, symbol,
+                );
```

Comment:
We shouldn't `panic` here because a weirdo/malicious SWF file can hit this. Sample:
[symbolClassConflict.zip](https://github.com/ruffle-rs/ruffle/files/11073224/symbolClassConflict.zip)

This SWF has 2 symbols that register to the same class. This actually works in the FP, and stage instances of either symbol will instantiate the class(!). Instantiating via AS `new Child()` creates the first registered child. So probably this should gracefully bail out if the class->symbol map already has a registered entry.

Reply:
I've changed this to log a warning, and added your SWF as a test case.